src/Sdk/RSWebApi/Contracts/AcquireJobRequest.cs

```diff
@@ -7,7 +7,29 @@ namespace GitHub.Actions.RunService.WebApi
     [DataContract]
     public class AcquireJobRequest
```

Comment:
This might be cleaner? 

```suggestion
    using System;
using System.Runtime.Serialization;

[DataContract]
public class AcquireJobRequest
{
    [DataMember(Name = "jobMessageId", EmitDefaultValue = false)]
    public string JobMessageID { get; set; }

    [DataMember(Name = "streamId", EmitDefaultValue = false)]
    private string DeprecatedJobMessageIdAlias { get; set; }

    [OnSerializing]
    private void OnSerializing(StreamingContext context)
    {
        DeprecatedJobMessageIdAlias = JobMessageID;
    }


    [OnDeserializing]
    private void OnDeserializing(StreamingContext context)
    {
        if (string.IsNullOrEmpty(JobMessageID) && !string.IsNullOrEmpty(DeprecatedJobMessageIdAlias))
        {
            JobMessageID = DeprecatedJobMessageIdAlias;
        }
    }
}

```

Reply:
I added unit tests to prove `streamId` is serialized in every scenario where `JobMessageID` has a non-empty value.

Reply:
```
[DataContract]
public class AcquireJobRequest
{
    [DataMember(Name = "jobMessageId", EmitDefaultValue = false)]
    public string JobMessageId { get; set; }

    [DataMember(Name = "streamId", EmitDefaultValue = false)]
    public string StreamId { get; set; }
}
```

Can we simply do this ☝️  and send both to service in this PR, dropping the one we don't want in the next runner release?

Reply:
+1, this is probably the right way: https://github.com/github/actions-run-service/pull/520#discussion_r1171204673

We can't depend on deployment order, for things like these, we need to provide full compat and then clean up next release

Reply:
I talked with Ting.  I'm still confident my proposed solution will work just as well, but am more interested in moving forward here, so if Ting's suggestion is the consensus best approach, so be it.

Reply:
In my mind the best path is: (1) deploy run service to accept both (2) runner just does the new way (3) after runner is release run service will only accept new things

Could you explain what you are thinking off from your side for posterity?